extensions/amp-fx-scroll-sync/0.1/scroll-sync-service.js

```diff
@@ -14,29 +14,99 @@
  * limitations under the License.
  */
 
-import {ScrollSyncEffect} from './scroll-sync-effect'
+import {ScrollSyncEffect} from './scroll-sync-effect';
+import {getService} from '../../../src/service';
+import {viewportFor} from '../../../src/viewport';
+import {vsyncFor} from '../../../src/vsync';
+
+let lastScrollTop_ = 0;
 
 class ScrollSyncService {
-  constructor() {
+
+  /**
+   * @param {!Window} win
+   */
+  constructor(win) {
+    this.win_ = win;
+
     /** @private @const {!Array<!ScrollSyncEffect>} */
     this.effects_ = [];
+
+    this.viewport_ = viewportFor(win);
+    this.vsync_ = vsyncFor(win);
+
+    this.viewport_.onScroll(() => {
+      this.vsync_.run({
```

Comment:
Scroll top is generally allowed to be read right from `onScroll`. But, of course, vsync is great as well. Just FYI.